workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/CodeWriter/Scopes/Abstract/ScopeBodyList.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Improbable.Gdk.CodeGeneration.CodeWriter.Scopes
+{
+    /// <summary>
+    /// A grouping of scoped bodies.
+    /// </summary>
+    public abstract class ScopeBodyList : ICodeBlock
+    {
+        private readonly List<ScopeBody> scopeBodies = new List<ScopeBody>();
+
+        protected ScopeBodyList(string declaration, Action<ScopeBody> populate)
```

Comment:
What does the declaration correspond to?

Reply:
```
<declaration>
{
	//populate action fills this in
}
```

Reply:
the `ScopeBodyList` is therefore

```
<declaration>
{
	//populate action fills this in
}
<declaration>
{
	//populate action fills this in
}
<declaration>
{
	//populate action fills this in
}
<declaration>
{
	//populate action fills this in
}
...
```

where the first scope is created with the constructor, and subsequent scopes are added through `AddScope`